client/src/apis/user.js

```diff
@@ -48,7 +50,7 @@ export const signup = async (user) => {
   try {
     const res = await axios.post("/api/users", user);
     setToken(res.data.token);
-    socketWithToken(res.data.token).emit("log in", res.data.user?._id);
+    importSocket().then((socket) => socket.emit("log in", res.data.user?._id));
```

Comment:
```suggestion
    const socket = await importSocket();
    socket.emit("log in", _id);
```

await looks better? 🤔 ...since this is in the async function